### ISSUE TITLE
Minimum distance between two octrees？

### ISSUE THREAD

**Issue by zichunxx** [isl-org/Open3D#5609]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### My Question

Is there a function that can be used to calculate the minimum distance between two sets of point clouds represented through  Octree?

In other words, is there a function that can be used to detect whether octrees are intersecting, i.e., collision detection.

Thanks in advance!
